go/vt/topo/k8stopo/server_flaky_test.go

```diff
@@ -79,7 +79,13 @@ func TestKubernetesTopo(t *testing.T) {
 	if err != nil {
 		t.Fatal("Unable to start k3s", err)
 	}
-	defer killK3s()
+	t.Cleanup(func() {
```

Comment:
This change makes tests less flaky by ensuring we give the OS time to fully shut down the k3s process. It's not perfect though because test timeouts can still leave a k3s process running in the background. So this is an improvement but still not perfect.